app/models/service_template.rb

```diff
@@ -538,4 +526,22 @@ def resource_actions_info
   def generic_custom_buttons
     CustomButton.buttons_for("Service")
   end
+
+  def adjust_service_type
+    return if service_type == self.class::SERVICE_TYPE_INTERNAL # internal cannot turn to composite
+
+    if service_resources.blank?
+      svc_type = self.class.SERVICE_TYPE_UNKNOWN
+    else
+      svc_type = self.class::SERVICE_TYPE_ATOMIC
+      service_resources.each do |sr|
+        if sr.resource_type == 'Service' || sr.resource_type == 'ServiceTemplate'
+          svc_type = self.class::SERVICE_TYPE_COMPOSITE
+          break
+        end
+      end
+    end
+
+    self.service_type = svc_type
```

Comment:
Should this save the record?

Reply:
No. It is a private method. The caller is responsible for saving the record.